hera_cal/vis_clean.py

```diff
@@ -1273,25 +1273,36 @@ def noise_eq_bandwidth(window, axis=-1):
 # ------------------------------------------
 
 
-def _filter_argparser():
+def _filter_argparser(parallelization_mode='file'):
     """
     Core Arg parser for commandline operation of hera_cal.delay_filter and hera_cal.xtalk_filter
-    
+    Parameters:
+        parallelization_mode : str, determines the arguments based on axis of parallelization.
+                               currently supports 'time' and 'baseline'.
+
     Returns:
         Argparser with core (but not complete) functionality that is called by _linear_argparser and
         _clean_argparser.
     """
     a = argparse.ArgumentParser(description="Perform delay filter of visibility data.")
-    a.add_argument("infilename", type=str, help="path to visibility data file to delay filter")
     a.add_argument("--filetype_in", type=str, default='uvh5', help='filetype of input data files (default "uvh5")')
     a.add_argument("--filetype_out", type=str, default='uvh5', help='filetype for output data files (default "uvh5")')
-    a.add_argument("--calfile", default=None, type=str, help="optional string path to calibration file to apply to data before delay filtering")
-    a.add_argument("--partial_load_Nbls", default=None, type=int, help="the number of baselines to load at once (default None means load full data")
     a.add_argument("--res_outfilename", default=None, type=str, help="path for writing the filtered visibilities with flags")
     a.add_argument("--clobber", default=False, action="store_true", help='overwrites existing file at outfile')
     a.add_argument("--spw_range", type=int, default=None, nargs=2, help="spectral window of data to foreground filter.")
     a.add_argument("--tol", type=float, default=1e-9, help='Threshold for foreground and xtalk subtraction (default 1e-9)')
-
+    if parallelization_mode == 'file':
+        a.add_argument("infilename", type=str, help="path to visibility data file to delay filter")
+        a.add_argument("--partial_load_Nbls", default=None, type=int, help="the number of baselines to load at once (default None means load full data")
+        a.add_argument("--calfile", default=None, type=str, help="optional string path to calibration file to apply to data before delay filtering")
+    else:
+        a.add_argument("--datafile_list", default=None, type=str, nargs='+', help="list of input data files")
+        a.add_argument("--calfile_list", default=None, type=str, nargs='+', help="list of input calibration files")
+        if parallelization_mode == 'baseline':
+            a.add_argument("--baseline_list", default=None, type=str, help="List of baselines to process."
```

Comment:
Clarify what this means for polarization.

Reply:
I've changed the mode to 'baseline-pol'. Users must also specify the polarizations to process (don't actually want to filter xy and yx if we aren't going to use them).